lib/elixir/lib/enum.ex

```diff
@@ -969,10 +969,18 @@ defmodule Enum do
     do_into(enumerable, collectable)
   end
 
+  def into(enumerable, %{__struct__: MapSet, map: map}) when map_size(map) == 0 do
```

Comment:
I am not a fan of the `map_size(map)` optimizations because it can make the code quite unpredictable if it depends on the right-side size. Can we get rid of those two clauses here for now or make them work for any size? Thank you.